server/src/main/java/io/crate/replication/logical/MetadataTracker.java

```diff
@@ -231,6 +232,11 @@ private CompletableFuture<?> processSubscription(String subscriptionName) {
             );
         }).exceptionallyCompose(err -> {
             var e = SQLExceptions.unwrap(err);
+            if (subscription.publications().size() > 1 && e instanceof PublicationUnknownException) {
+                // There are other publications to follow, keep tracking
```

Comment:
wanted to clarify before I do - do we need to update subscription metadata on a subscriber? On the one hand, it causes publication not found error on a publisher, on the other hand, -1 cluster update on a subscriber

Reply:
If we just re-schedule on such an exception, the updates from the other publications are never processed. It will again fail on the next schedule. I think we must handle this case inside the publication state action to not throw an exception, probably configured by a request flag. I'd say we should keep the publication at the subscription metadata, so if the missing publication is re-created, the replication can continue. But not completely sure about that. @mfussenegger wdyt?

Reply:
> It will again fail on the next schedule

oh, I somehow thought it's harmless for other pubs, so current draft is defo wrong, thx for clarification!

>missing publication is re-created, the replication can continue

but then it will fail with table already exists as tables were replicated. [dynamic check](https://github.com/crate/crate/pull/12342) for detecting self-subscription loops will detect it and fail in that case as well.

another reason is that user can re-create publication with different set of tables and means that it's completely different setup but accidentally choses name of the dropped publication - and might be surprised that it fails if new setup partially overlaps with old setup.

I think I have to update subscriber's publications list as well.

Reply:
Maybe instead of raising an error in the publications state action we include unavailable publications in the response?

Then the subscriber can decide what do do (and should probably also update the subscription state with some information that one or more publications are unavailable)

Reply:
>we include unavailable publications in the response?

sounds good. also can fold this cluster update into existing one, which we do right after the remote state fetch.